.github/workflows/validation-infrastructure-tests.yml

```
@@ -0,0 +1,233 @@
+      matrix:
+        python-version: ['3.10', '3.11', '3.12']
+        test-suite: ['unit', 'integration']
+
+    name: Python ${{ matrix.python-version }} - ${{ matrix.test-suite }}
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v4
```

actionlint (1.7.7)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)